lib/whitehall_importer/create_revision.rb

```diff
@@ -115,13 +115,21 @@ def tags(associations)
 
     def create_image_revisions(images)
       images.reduce([]) do |memo, image|
-        memo << WhitehallImporter::CreateImageRevision.call(record, image, memo.map(&:filename))
+        already_imported = WhitehallImportedAsset.find_by(original_asset_url: image["url"])
+        unless already_imported
+          memo << WhitehallImporter::CreateImageRevision.call(record, image, memo.map(&:filename))
+        end
+        memo
```

Comment:
Oh neat - I see that this will save us adding superfluous images and attachments which is super cool. Not sure if my suggestion of asset record per file complicates that, hopefully there's a way.

Do we use this create_image_revisions as the means of getting the array of all the image revisions of the document? Assuming so, I believe by skipping items we'll be missing them from the array of image revisions so we may need the means to look up ones that already exist. So perhaps this become more of a `find_or_create_image_revisions`?

Minor but this can be a little cleaner as 

```
        already_imported = WhitehallImportedAsset.find_by(original_asset_url: image["url"])
        next memo if already_imported
        memo << WhitehallImporter::CreateImageRevision.call(record, image, memo.map(&:filename))
```

Reply:
Good point. Have now changed it to `find_or_create_image_revisions`.